core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -596,60 +588,61 @@ default Logger makeContextAware(Logger logger) {
     }
 
     /**
-     * Registers {@code callback} to be run when re-entering this {@link RequestContext}, usually when using
-     * the {@link #makeContextAware} family of methods. Any thread-local state associated with this context
-     * should be restored by this callback.
+     * Do not use this method. The callback is never invoked.
      *
-     * @param callback a {@link Consumer} whose argument is this context
-     */
-    void onEnter(Consumer<? super RequestContext> callback);
-
-    /**
-     * Registers {@code callback} to be run when re-entering this {@link RequestContext}, usually when using
-     * the {@link #makeContextAware} family of methods. Any thread-local state associated with this context
-     * should be restored by this callback.
+     * <p>This method has a limitation which is that the callback cannot be called when this
+     * {@link RequestContext} is pushed first. So we decided to remove this method and provide
+     * better API after 1.0. If you are using this, please find another way to achieve the same behavior.
+     * Most of the cases, you might be able to just use {@link RequestLogListener}.
+     * Also, you might follow this approach which is using your own thread-local.
```

Comment:
I guess it's opposite, uses `RequestContext` instead of own thread-local.

Maybe looking at the PR is too confusing to make it worth linking to.